### DISCUSSION TITLE
Error when adding `muiCache` from mui-5 documentation

### DISCUSSION THREAD

**Discussion by nenaddimitrijevic89** [garronej/tss-react#64] [Q&A] [answered]
Does someone had this kind of error:
```
Argument of type '{ key: string; prepend: boolean; }' is not assignable to parameter of type 'Options'.
  Object literal may only specify known properties, and 'prepend' does not exist in type 'Options'
```
when adding this piece of code from mui-5 documentation

`export const muiCache = createCache({
    key: 'mui',
    prepend: true
})`

**Comment by garronej** [comment, marked as answer]
Hi @nenaddimitrijevic89,  
Make sure you are importing createCache like this: `import createCache from "@emotion/cache";`.  
If you are, update `@emotion/styled` and `@emotion/react` to the latest version and you should be good to go.  

Best

**Reply by nenaddimitrijevic89** [reply]
@garronej thank you for answering :) 
but it is not a problem, I still get the same error `prepend ` doesnt exist in type `Options`

**Reply by garronej** [reply]
What version of @emotion/cache do you have installed?

**Reply by nenaddimitrijevic89** [reply]
> What version of @emotion/cache do you have installed?

now its ok, `@emotion/cache` needs to be installed to latest along with these two `@emotion/styled` and `@emotion/react`.
@garronej thanks, you saved me 👍

**Reply by garronej** [reply]
You are welcome. I don't understand why you had to install it manually. What package manager are you using?

**Reply by garronej** [reply]
I would be interested to see the lock file before you installed emotion/cache manually

**Reply by nenaddimitrijevic89** [reply]
> I would be interested to see the lock file before you installed emotion/cache manually

I am using yarn @garronej

**Reply by garronej** [reply]
Yarn classic (aka Yarn 1) or Yarn Berry (aka Yarn 2) ?  
Do you have a node module in your repo?  
Is your `yarn.lock` a sensible information? If not I would be very interested in seeing it.  I would be interested to see why you had to explicitly install `@emotion/cache`.  
I saw you marked as answered my answer suggesting to upgrade `@emotion/react` and `@emotion/styled` but if I am not mistaking what actually solved the problem for you was to `yarn add @emotion/cache`.

**Reply by nenaddimitrijevic89** [reply]
its Yarn 1 that I am using.
`yarn add @emotion/cache` actually solved my problem.
this is difference in `yarn.lock` before and after i install `@emotion/cache`:
![yarn lock](https://user-images.githubusercontent.com/60787476/156559800-c52350b8-af6d-45c4-a34c-768dd8192366.PNG)

**Reply by garronej** [reply]
Thank you for your answer!  
I see there is another addition further down can I see it as well? 

![image](https://user-images.githubusercontent.com/6702424/156561049-26fdce7f-e57d-4784-a388-92e0da730079.png)

**Reply by nenaddimitrijevic89** [reply]
You are welcome. @garronej 
here is:
![yarn](https://user-images.githubusercontent.com/60787476/156561616-445bb523-7051-4c2b-b9cf-c90af0f685a1.PNG)

**Reply by garronej** [reply]
Thank you very much and sorry to ask so much of you.  
It seems you hadn't `tss-react` in your `yarn.lock` before 🤔   
Having `@emotion/cache` explicitly in your dependency is something that might cause you problem down the drain. 
Could you remove `@emotion/cache` from your `package.json` and install your dependencies again.  
If you have `tss-react`, `@mui/material`, `@emotion/react` and `@emotion/styled` pointing yo the latest versions in your `package.json` you shouldn't need  `@emotion/cache`.

**Reply by nenaddimitrijevic89** [reply]
@garronej bro, its not working without manually installed `@emotion/cache`. I tried this now.
Guess I must have `@emotion/cache` installed as dependancy.

**Reply by garronej** [reply]
Well, all I can say for sure is that you have emotion v10 (more than two years old) and emotion v11 cohabiting in your project.  
It can be the case if you are using Storybook but it doesn’t explain why `@emotion/cache` resolves to v11 instead of v10.  
Could be that  you have `"@emotion/xxx": "^10.X.Y"` instead of `@emotion/yyy": "^11.X.Y` in your `package.json` or it could be something else that I don't understand but I can't figure it out without seeing the full `package.json` and `yarn.lock`.  

Anyway, 
Best regards,

**Reply by nenaddimitrijevic89** [reply]
@garronej here are my `yarn.lock` and `package.json`.
Thanks for your help :)
[yarn.zip](https://github.com/garronej/tss-react/files/8185634/yarn.zip)

**Reply by nenaddimitrijevic89** [reply]
> @garronej here are my `yarn.lock` and `package.json`. Thanks for your help :) [yarn.zip](https://github.com/garronej/tss-react/files/8185634/yarn.zip)

Hi @garronej, can you help about this, i attached  `yarn.lock` and `package.json`. Can you see why i had to manually install `@emotion/cache` ?
Thank you in advance

**Reply by garronej** [reply]
You have `react-select` that d

You are using [`@storybook/addon-knobs`](https://www.npmjs.com/package/@storybook/addon-knobs) that is a deprecated package that uses `react-select` that has a hard dependency on emotion 10.  
If you can, get rid of Storybook Addon Knobs in favor of `@storybook/addon-controls`.  

If you cant add a theses to the resolution field of your package.json:  
```json
"resolutions": {
        "babel-loader": "8.2.2",
        "@emotion/cache": "11.7.1",
        "@emotion/serialize": "1.0.2",
        "@emotion/utils": "1.1.0"
}
```

I asume you have done this already but you should also follow the instructions to make Storybook work with emotion 11: https://docs.tss-react.dev/readme-1
![image](https://user-images.githubusercontent.com/6702424/157139963-54e41c3e-28c4-4c55-b0f3-0c8f70cbe249.png)  

Regards
